packages/hash/frontend/src/components/layout/PageHeader/SearchBar/SearchInput.tsx

```diff
@@ -0,0 +1,144 @@
+import { Box, InputAdornment, InputBase, Tooltip } from "@mui/material";
+import { useRef } from "react";
+import { useKeys } from "rooks";
+import { faXmark } from "@fortawesome/free-solid-svg-icons";
+
+import { SearchIcon, FontAwesomeSvgIcon } from "../../../icons";
+
+export const SearchInput: React.FC<{
+  displayedQuery: string;
+  isMobile: boolean;
+  setResultListVisible: (visible: boolean) => void;
+  setQueryText: (queryText: string) => void;
+}> = ({ displayedQuery, isMobile, setResultListVisible, setQueryText }) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useKeys(["AltLeft", "KeyK"], () => inputRef.current?.focus());
+
+  return (
+    <Box
+      sx={(theme) => ({
+        position: "relative",
+        borderRadius: theme.spacing(0.75),
+        marginRight: theme.spacing(2),
+        border: `1px solid ${theme.palette.gray[30]}`,
+        width: isMobile ? "100%" : undefined,
+        ":focus": {
+          margin: "-1px",
+          border: `2px solid ${theme.palette.blue[70]}`,
+        },
```

Comment:
1) I personally feel we should be more concerned about type safety when trying to access our theme properties (color/typography) than common CSS style rules.
2) For scenarios like the one described, I agree that it makes sense to just list the styles that we need to override
```
border: `1px solid transparent`
":focus": {
  borderColor: ({ palette }) => palette.blue[70]
}
```

I'd say we can write border styles in the short form initially, but when dealing with hover/focus states where a particular style changes, then we should probably just list that style

Reply:
> I personally feel we should be more concerned about type safety when trying to access our theme properties (color/typography) than common CSS style rules.

@teenoh are there scenarios where we aren't doing this in a type-safe way right now?

Reply:
> I'd say we can write border styles in the short form initially, but when dealing with hover/focus states where a particular style changes, then we should probably just list that style

To be clear we can do this, this isn't a big issue I'm concerned about. I'm just trying to point out that we're missing out on type-safety we could be getting otherwise, just for the convenience of defining something in 1 line instead of 3 lines - which in my opinion is a bad tradeoff.

Especially given that invalid CSS is hard for us to catch at the moment.

Reply:
Yeah @benwerner01, did a quick search in the BP repo, and this repo and we make use of the `border` shorthand approach (`1px solid ${color}`) more times than the long approach, the same thing happens when we are setting `outline` for focus styles. I feel it's common for people to set borders this way and don't think it will be difficult to catch invalid CSS applied to the border since we can debug that by checking dev tools.

Reply:
> I feel it's common for people to set borders this way and don't think it will be difficult to catch invalid CSS applied to the border since we can debug that by checking dev tools.

I suppose we'll have to agree to disagree, I think an invalid border definition is going to be hard to spot in a PR review that includes significant styling. Consider:

```TypeScript
  border: `1px solid transparant`,
```

I think there's a 75% chance I wouldn't catch that.

I also don't think it's reasonable to expect PR reviewers to go through all the added styling the in the dev console to check for invalid CSS.

Reply:
> You may also already know, but some properties like transform are not completely type-safe, so we can't fully rely on TS to ensure all properties have valid values.

Yeah, I'm aware we can't make all our CSS type safe - I just think we should make use of type safety where we can.

To be clear I don't think we should block a PR from merging if it uses a shorthand, the only reason I brought this up here to make sure we're aware of additional type-safety we can make use of.